### PR TITLE
feat: add Python review and roadmap support

### LINKED ISSUE #32: [Feature]: Add Python review support for requirements.txt and pyproject.toml
## Problem statement
Python support exists for `analyze`, but `review` still only supports `package.json`-style workflows. This leaves the most important PR-review path incomplete for Python users and keeps the feature in an experimental state.

## Proposed solution
Add Python review support for `requirements.txt` and `pyproject.toml`.

- detect and compare Python dependency changes between base/head
- reuse the current Markdown review output shape where possible
- support practical review flows in mixed repos without regressing npm behavior
- define the minimum supported semantics for pinned vs ranged Python dependencies

## Alternatives considered
Keep Python support limited to `analyze` only. This would reduce implementation work, but it leaves the user-facing feature set inconsistent and weakens the release story for the next minor version.

## Example CLI or Action usage
```bash
npx aminet review requirements.txt --base HEAD~1 --security
npx aminet review pyproject.toml --base origin/main --security
```

### LINKED ISSUE #33: [Feature]: Generalize GitHub Action inputs and docs for Python manifests
## Problem statement
The GitHub Action and README are still described primarily in terms of `package.json`, even though the project now supports Python analysis. That mismatch creates incorrect expectations and makes the Python path harder to discover.

## Proposed solution
Generalize the GitHub Action inputs and documentation for Python manifests.

- update `action.yml` input descriptions so `path` is not npm-only
- add Python-oriented examples to README and release docs
- verify argument forwarding remains correct for both npm and Python manifests
- make it clear which capabilities are analyze-only vs review-capable

## Alternatives considered
Leave the implementation ahead of the docs. This keeps the short-term work smaller, but it increases support burden and makes the action feel incomplete.

## Example CLI or Action usage
```yaml
- uses: gorira-tatsu/aminet@v0.x.y
  with:
    path: requirements.txt
    security: "true"
```

### LINKED ISSUE #34: [Feature]: Harden pyproject.toml dependency parsing for real-world project layouts
## Problem statement
`pyproject.toml` parsing currently focuses on a narrow subset of layouts. Real projects often use optional dependencies, multiline arrays, additional sections, and formatting patterns that can make regex-based parsing fragile.

## Proposed solution
Harden `pyproject.toml` dependency parsing for real-world project layouts.

- expand fixture coverage for realistic `pyproject.toml` files
- support more valid dependency declarations under supported scopes
- reduce false negatives caused by formatting details
- document the exact supported subset when full TOML coverage is still out of scope

## Alternatives considered
Keep the parser minimal and rely on docs to explain constraints. That reduces implementation cost, but it makes Python support feel unreliable in practice.

## Example CLI or Action usage
```bash
npx aminet analyze pyproject.toml
npx aminet review pyproject.toml --base HEAD~1
```

### LINKED ISSUE #35: [Feature]: Improve Python best-effort resolution warnings and review messaging
## Problem statement
Python range specifiers currently fall back to best-effort resolution, and unsupported markers can be skipped. Those behaviors are acceptable for an MVP, but the current messaging is not strong enough for users to understand confidence levels.

## Proposed solution
Improve Python best-effort resolution warnings and review messaging.

- make best-effort resolution explicit in CLI and review output
- surface skipped marker-based dependencies with actionable explanations
- distinguish unsupported input from successfully analyzed input
- keep messages concise enough for PR review comments

## Alternatives considered
Leave the current warnings as-is. This avoids output churn, but it risks users over-trusting incomplete Python analysis.

## Example CLI or Action usage
```bash
npx aminet analyze requirements.txt --security
npx aminet review pyproject.toml --base HEAD~1
```

### LINKED ISSUE #36: [Feature]: Add regression coverage for Python review flows and mixed-repo fixtures
## Problem statement
Python support was added recently, but regression coverage is still light around review flows, mixed repositories, and action-level argument forwarding. That creates a high risk of silent breakage as the feature evolves.

## Proposed solution
Add regression coverage for Python review flows and mixed-repo fixtures.

- add tests for Python review diffs and Markdown output
- add mixed npm/Python repo fixtures where relevant
- verify GitHub Action argument forwarding for Python manifests
- protect current npm behavior while expanding Python coverage

## Alternatives considered
Rely on manual testing during feature work. That is faster in the short term, but it is not enough for a release-defining feature.

## Example CLI or Action usage
```bash
pnpm test
```

### LINKED ISSUE #37: [Feature]: Define and document Python lockfile support strategy (poetry/pdm/uv)
## Problem statement
The roadmap references `poetry.lock`, `pdm.lock`, and `uv.lock`, but there is no explicit product decision yet on scope, order, or compatibility expectations. That ambiguity makes it harder to stage the Python roadmap cleanly.

## Proposed solution
Define and document the Python lockfile support strategy.

- decide which lockfile formats are first-class targets
- define whether support means `analyze`, `review`, or both
- capture sequencing and compatibility constraints in docs or roadmap notes
- use the strategy to guide follow-up implementation issues

## Alternatives considered
Start implementing lockfiles opportunistically. That may move faster at first, but it increases the risk of partial support with unclear boundaries.

## Example CLI or Action usage
```bash
# roadmap / design task, no direct CLI example yet
```

### LINKED ISSUE #38: [Feature]: Improve init-generated config for private registry workflows
## Problem statement
`aminet init` can generate a baseline config, but it does not yet strongly guide users through private-registry workflows that are now part of the intended product surface.

## Proposed solution
Improve init-generated config for private registry workflows.

- review which config keys should be generated by default
- improve guidance around `excludePackages` and related settings
- make the output more useful for real team setups without embedding secrets
- ensure generated config stays aligned with current CLI behavior

## Alternatives considered
Keep `init` minimal and rely on README examples. That is simpler, but it underuses the feature and leaves operational setup fragmented.

## Example CLI or Action usage
```bash
npx aminet init
npx aminet init --defaults --merge
```

### LINKED ISSUE #39: [Feature]: Document private registry auth and exclude-packages usage across CLI and Action
## Problem statement
Private registry support exists, but the relationship between `npm-token` and `exclude-packages` is still easy to misunderstand across CLI and Action usage. That confusion directly affects adoption in real repositories.

## Proposed solution
Document private registry auth and `exclude-packages` usage across CLI and Action flows.

- add clear examples for authenticated private packages
- add clear examples for intentionally skipped internal packages
- explain when to prefer auth vs exclusion
- keep README, action docs, and generated config guidance aligned

## Alternatives considered
Treat the current docs as sufficient. That avoids more documentation work, but it leaves a recently added feature under-explained.

## Example CLI or Action usage
```bash
NPM_TOKEN=xxx npx aminet review package.json --base HEAD~1
npx aminet review package.json --base HEAD~1 --exclude-packages "@scope/*"
```

### LINKED ISSUE #40: [Feature]: Improve degraded-mode UX when persistent cache is unavailable
## Problem statement
When persistent cache support is unavailable, the tool degrades safely, but the UX still feels like a fallback path rather than an intentional operational mode. That can make the tool seem less reliable than it is.

## Proposed solution
Improve degraded-mode UX when persistent cache is unavailable.

- review cache-related warnings and informational messages
- make fallback behavior explicit without sounding like a fatal condition
- ensure cache subcommands remain coherent in degraded mode
- document practical expectations for environments without native bindings

## Alternatives considered
Keep the current behavior and messaging. That is acceptable technically, but it leaves avoidable friction in constrained environments.

## Example CLI or Action usage
```bash
npx aminet cache stats
npx aminet analyze package.json --no-cache
```

### LINKED ISSUE #41: [Feature]: Add poetry.lock, pdm.lock, and uv.lock support
## Problem statement
The roadmap identifies lockfile support as the main path from experimental Python support toward broader ecosystem credibility, but the implementation does not yet exist.

## Proposed solution
Add support for `poetry.lock`, `pdm.lock`, and `uv.lock`.

- parse supported lockfile formats
- resolve dependency versions from lockfiles where available
- define whether support lands in `analyze`, `review`, or staged phases
- add fixtures and regression coverage for each format that is accepted

## Alternatives considered
Continue relying only on manifest-level parsing. That keeps the implementation small, but it limits Python accuracy and prevents the roadmap from advancing meaningfully.

## Example CLI or Action usage
```bash
npx aminet analyze poetry.lock
npx aminet analyze uv.lock
```

### LINKED ISSUE #42: [Feature]: Unify npm and PyPI output semantics across analyze/review/reporting
## Problem statement
As npm and PyPI support expand, differences in analyze/review/report semantics can become harder to reason about. Without explicit convergence work, the product may feel like two loosely related tools.

## Proposed solution
Unify npm and PyPI output semantics across analyze, review, and reporting.

- define where output should be ecosystem-agnostic
- align confidence messaging and unsupported-case handling
- keep the review comment shape understandable across ecosystems
- identify intentional ecosystem-specific differences rather than accidental ones

## Alternatives considered
Let each ecosystem evolve independently. That may be faster short-term, but it increases maintenance cost and makes a 1.0 contract harder to define.

## Example CLI or Action usage
```bash
npx aminet analyze package.json --json
npx aminet analyze pyproject.toml --json
```

### LINKED ISSUE #43: [Feature]: Define v1.0 release criteria, compatibility guarantees, and graduation checklist
## Problem statement
The project is still pre-1.0, and the roadmap needs an explicit definition of what "1.0-ready" means. Without release criteria and compatibility rules, issue prioritization will stay ambiguous.

## Proposed solution
Define the v1.0 release criteria, compatibility guarantees, and graduation checklist.

- define the minimum supported workflows and ecosystems for 1.0
- define CLI / output stability expectations
- define documentation and test thresholds required before 1.0
- use the checklist to separate must-have work from desirable follow-ons

## Alternatives considered
Decide 1.0 criteria only when the backlog gets smaller. That postpones planning debt, but it weakens milestone discipline and makes issue prioritization noisier.

## Example CLI or Action usage
```bash
# roadmap / release-planning task
```

### PR DESCRIPTION
**Pull request by gorira-tatsu**
## Summary
- add Python review support for `requirements.txt` and `pyproject.toml`
- add Python lockfile parsing for `poetry.lock`, `pdm.lock`, and `uv.lock`
- align CLI, GitHub Action, workflow, docs, and roadmap criteria around the expanded Python support

## Testing
- `pnpm lint`
- `pnpm build`
- `pnpm test`

Closes #32
Closes #33
Closes #34
Closes #35
Closes #36
Closes #37
Closes #38
Closes #39
Closes #40
Closes #41
Closes #42
Closes #43

## Summary by CodeRabbit

* **New Features**
  * Added Python ecosystem support for dependency review and analysis, including `requirements.txt`, `pyproject.toml`, and Python lockfiles (`poetry.lock`, `pdm.lock`, `uv.lock`).
  * Extended GitHub Actions workflow to trigger on Python dependency file changes.

* **Documentation**
  * Updated README with Python manifest examples and configuration guidance.
  * Added ROADMAP.md documenting project milestones and release criteria.

* **Chores**
  * Improved CLI error messaging for cache unavailability and private registry guidance.

### CODE REVIEW

File: .github/workflows/ami-review.yml [hunk 1]
```diff
@@ -4,9 +4,14 @@ on:
...
     paths:
       - "package.json"
+      - "requirements.txt"
+      - "pyproject.toml"
       - "pnpm-lock.yaml"
       - "package-lock.json"
       - "bun.lock"
+      - "poetry.lock"
+      - "pdm.lock"
+      - "uv.lock"
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply .github/workflows/ami-review.yml:14]: Fixed. The workflow now detects the changed manifest from the PR diff and passes that manifest, plus an adjacent lockfile when applicable, into the action instead of always hard-coding `package.json`.

File: src/core/lockfile/python-parser.ts [hunk 2]
```diff
@@ -183,20 +248,35 @@ function extractTomlArray(content: string, section: string, field: string): stri
...
+  return [...packages.entries()]
+    .filter(([, spec]) => isBestEffortVersionSpec(spec))
+    .map(([name]) => name)
+    .sort();
+}
+
+function isBestEffortVersionSpec(versionSpec: string): boolean {
+  const trimmed = versionSpec.trim();
+  if (trimmed === "" || trimmed.toLowerCase() === "latest") return true;
+  return !/^\d+(\.\d+){0,2}([A-Za-z0-9._+-]+)?$/.test(trimmed);
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply src/core/lockfile/python-parser.ts]: Fixed. I widened the exact-version matcher so pinned PEP 440 versions like `1.2.3.4` are no longer mislabeled as best-effort.

File: src/core/lockfile/parser.ts [hunk 3]
```diff
@@ -216,3 +234,26 @@ function normalizePnpmVersion(version: string | undefined): string | null {
...
+    const packages = new Map<string, string>();
+    const blocks = content.split(/\n(?=\[\[package\]\])/g);
+
+    for (const block of blocks) {
+      if (!block.includes("[[package]]")) continue;
+      const nameMatch = block.match(/^\s*name\s*=\s*["']([^"']+)["']/m);
+      const versionMatch = block.match(/^\s*version\s*=\s*["']([^"']+)["']/m);
+      if (!nameMatch || !versionMatch) continue;
+      packages.set(nameMatch[1], versionMatch[1]);
+    }
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply src/core/lockfile/parser.ts]: Fixed. The Python lockfile parser now splits on `\r?\n`, so CRLF-separated `poetry.lock` / `pdm.lock` / `uv.lock` files are parsed correctly.

File: src/core/lockfile/parser.ts [hunk 4]
```diff
@@ -9,20 +9,28 @@ export interface LockfileEntry {
...
  * Try to find and parse a lockfile adjacent to the given package.json path.
  * Returns null if no lockfile found.
  */
-export async function tryParseLockfile(packageJsonPath: string): Promise<LockfileResult | null> {
-  const dir = packageJsonPath.replace(/[/\\][^/\\]+$/, "");
+export async function tryParseLockfile(
+  manifestPath: string,
+  ecosystem: "npm" | "pypi" = "npm",
+): Promise<LockfileResult | null> {
+  const dir = manifestPath.replace(/[/\\][^/\\]+$/, "");
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply src/core/lockfile/parser.ts:30]: Fixed. I updated the JSDoc to refer to an adjacent manifest path rather than `package.json` specifically.

File: src/index.ts [hunk 5]
```diff
@@ -19,7 +19,7 @@ program
   .description("Analyze dependencies, licenses, and vulnerabilities")
   .argument(
     "<package>",
-    "Package (e.g., express@4.21.2) or file path (package.json, requirements.txt)",
+    "Package (e.g., express@4.21.2) or file path (package.json, pyproject.toml, poetry.lock)",
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply src/index.ts]: Fixed. The `analyze` help text now lists the full supported Python inputs: `requirements.txt`, `pyproject.toml`, `poetry.lock`, `pdm.lock`, and `uv.lock`.

File: src/cli/commands/review.ts [hunk 6]
```diff
@@ -267,12 +372,17 @@ function toGitRelativePath(filePath: string, gitRoot: string | null): string {
...
+    ? ["uv.lock", "poetry.lock", "pdm.lock"]
+    : ["pnpm-lock.yaml", "bun.lock", "package-lock.json"];
+}
 
 export async function loadAdjacentLockfile(
   packageJsonPath: string,
   ref?: string,
   explicitLockfilePath?: string,
+  ecosystem: "npm" | "pypi" = "npm",
 ): Promise<ReturnType<typeof parseLockfile> | null> {
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply src/cli/commands/review.ts:386]: Fixed. I renamed the parameter to `manifestPath` and updated the related local variable names to make the helper ecosystem-agnostic.

File: .github/workflows/ami-review.yml [hunk 7]
```diff
@@ -4,9 +4,14 @@ on:
   pull_request:
     paths:
       - "package.json"
+      - "requirements.txt"
+      - "pyproject.toml"
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply .github/workflows/ami-review.yml:8]: Fixed. The workflow now resolves the changed review manifest dynamically from the PR diff, so Python-only PRs no longer run the action against `package.json`.

File: src/core/lockfile/python-parser.ts [hunk 8]
```diff
@@ -1,170 +1,235 @@
...
-  // Extract [project.optional-dependencies].dev array
-  const devDepsArray = extractTomlArray(content, "project.optional-dependencies", "dev");
-  for (const dep of devDepsArray) {
-    addParsedDep(dep, devDependencies);
+  parsePoetryDependencySection(content, "tool.poetry.dependencies", dependencies, skipped);
+  for (const group of DEV_LIKE_GROUPS) {
+    parsePoetryDependencySection(
+      content,
+      `tool.poetry.group.${group}.dependencies`,
+      devDependencies,
```
[the remark this answers is not in the document]
Comment (gorira-tatsu) [reply src/core/lockfile/python-parser.ts:121]: Fixed. I added support for legacy `[tool.poetry.dev-dependencies]` so `--dev` reviews do not silently drop those packages.

**Comment by gorira-tatsu** [comment]
Addressed the automated review feedback in `c73f5fb`.

- Switched the example workflow to detect the changed manifest dynamically, including Python manifests and adjacent lockfiles.
- Added legacy Poetry `dev-dependencies` support and tightened exact-version detection for Python best-effort notes.
- Fixed the CRLF Python lockfile parsing edge case and cleaned up the ecosystem-agnostic naming/help text.

Validation after the changes:
- `pnpm lint`
- `pnpm build`
- `pnpm test`
